lib/data_source/client/github.rb

```diff
@@ -95,8 +163,8 @@ def write_cache(filename, value)
         return value unless @cache_dir
 
         FileUtils.mkdir_p(@cache_dir) unless File.directory?(@cache_dir)
-        path = File.join(@cache_dir, filename)
-        File.open(path, 'w') { |f| f.write(value) }
+        path = File.join(@cache_dir, @cache_name_hash[filename])
+        File.open(path, 'w') { |f| f.write(JSON.dump(value)) }
```

Comment:
I originally passed the `response.body` directly to avoid parsing twice. Do you think there could be another way to keep that and at the same time add support for your pagination changes? That'd be neat!

Reply:
I did think about that, but since the cash is going to hold multiple responses, the neatest thing is to leverage some structured data (i.e.JSON).